Sources/SecureEnclaveValet.swift

```diff
@@ -0,0 +1,285 @@
+            
+        } else {
+            let valet = SecureEnclaveValet(identifier: identifier, accessControl: accessControl)
+            identifierToValetMap.setObject(valet, forKey: key)
+            return valet
+        }
+    }
+    
+    /// - parameter identifier: A non-empty string that must correspond with the value for keychain-access-groups in your Entitlements file.
+    /// - returns: A SecureEnclaveValet that reads/writes keychain elements that can be shared across applications written by the same development team.
+    public class func sharedAccessGroupValet(with identifier: Identifier, accessControl: SecureEnclaveAccessControl) -> SecureEnclaveValet {
+        let key = Service.sharedAccessGroup(identifier, .secureEnclave(accessControl)).description as NSString
+        if let existingValet = identifierToValetMap.object(forKey: key) {
+            return existingValet
+            
+        } else {
+            let valet = SecureEnclaveValet(sharedAccess: identifier, accessControl: accessControl)
+            identifierToValetMap.setObject(valet, forKey: key)
+            return valet
+        }
+    }
+    
+    // MARK: Equatable
+    
+    /// - returns: `true` if lhs and rhs both read from and write to the same sandbox within the keychain.
+    public static func ==(lhs: SecureEnclaveValet, rhs: SecureEnclaveValet) -> Bool {
+        return lhs.service == rhs.service
+    }
+    
+    // MARK: Private Class Properties
+    
+    private static let identifierToValetMap = NSMapTable<NSString, SecureEnclaveValet>.strongToWeakObjects()
+    
+    // MARK: Initialization
+    
+    @available(*, deprecated)
+    public override init() {
+        fatalError("Do not use this initializer")
+    }
+    
+    private init(identifier: Identifier, accessControl: SecureEnclaveAccessControl) {
+        service = .standard(identifier, .secureEnclave(accessControl))
+        keychainQuery = service.generateBaseQuery()
+        self.identifier = identifier
+        self.accessControl = accessControl
+    }
+    
+    private init(sharedAccess identifier: Identifier, accessControl: SecureEnclaveAccessControl) {
+        service = .sharedAccessGroup(identifier, .secureEnclave(accessControl))
+        keychainQuery = service.generateBaseQuery()
+        self.identifier = identifier
+        self.accessControl = accessControl
+    }
+    
+    // MARK: Hashable
+    
+    public override var hashValue: Int {
+        return service.description.hashValue
+    }
+    
+    // MARK: Public Properties
+    
+    public let identifier: Identifier
+    @objc
+    public let accessControl: SecureEnclaveAccessControl
+    
+    // MARK: Public Methods
+    
+    /// - returns: `true` if the keychain is accessible for reading and writing, `false` otherwise.
+    /// - note: Determined by writing a value to the keychain and then reading it back out. Will never prompt the user for Face ID, Touch ID, or password.
+    @objc
```

Comment:
Is this necessary here? (Same with on `accessControl` above)

Reply:
Yup! See https://github.com/apple/swift-evolution/blob/master/proposals/0160-objc-inference.md re why we need to explicitly expose to `@objc` now :)